commit 5dc4d807070a357720e8567406b170286b4cc604
Author: Adrian Schollmeyer <git@nexadn.de>
Date:   Mon Aug 14 17:52:20 2023 +0200

    daemon: make compatible with native non-header-only httplib
    
    cpp-httplib can be installed as a non-header-only system library, which
    breaks this software as it relies on internal API which is not available
    in the non-header-only version of cpp-httplib. This commit imports the
    declaration from the private API so the software works with a native
    httplib.

--- a/daemon/CMakeLists.txt
+++ b/daemon/CMakeLists.txt
@@ -20,9 +20,7 @@ if (NOT RAVENNNA_ALSA_LKM_DIR)
 endif()
 
 # use sysroot cpp-http lib unless one was explicitly provided  in cmdline
-if (NOT CPP_HTTPLIB_DIR)
-    find_path( CPP_HTTPLIB_DIR "httplib.h" REQUIRED)
-endif()
+find_package(httplib REQUIRED)
 
 find_library(AVAHI_LIBRARY-COMMON NAMES avahi-common)
 find_library(AVAHI_LIBRARY-CLIENT NAMES avahi-client)
@@ -31,7 +29,7 @@ set(AVAHI_LIBRARIES ${AVAHI_LIBRARY-COMMON} ${AVAHI_LIBRARY-CLIENT})
 set(AVAHI_INCLUDE_DIRS ${AVAHI_INCLUDE_DIR})
 
 find_package(Boost COMPONENTS system thread log program_options REQUIRED)
-include_directories(aes67-daemon ${RAVENNA_ALSA_LKM_DIR}/common ${RAVENNA_ALSA_LKM_DIR}/driver ${CPP_HTTPLIB_DIR} ${Boost_INCLUDE_DIR})
+include_directories(aes67-daemon ${RAVENNA_ALSA_LKM_DIR}/common ${RAVENNA_ALSA_LKM_DIR}/driver ${Boost_INCLUDE_DIR})
 add_definitions( -DBOOST_LOG_DYN_LINK -DBOOST_LOG_USE_NATIVE_SYSLOG )
 add_compile_options( -Wall )
 set(SOURCES error_code.cpp json.cpp main.cpp session_manager.cpp http_server.cpp config.cpp interface.cpp log.cpp sap.cpp browser.cpp rtsp_client.cpp mdns_client.cpp mdns_server.cpp rtsp_server.cpp utils.cpp)
@@ -49,7 +47,7 @@ if(ENABLE_TESTS)
     add_subdirectory(tests)
 endif()
 
-target_link_libraries(aes67-daemon ${Boost_LIBRARIES})
+target_link_libraries(aes67-daemon ${Boost_LIBRARIES} httplib::httplib)
 if(WITH_AVAHI)
   MESSAGE(STATUS "WITH_AVAHI")
   add_definitions(-D_USE_AVAHI_)
--- a/daemon/utils.hpp
+++ b/daemon/utils.hpp
@@ -57,4 +57,9 @@ struct SDPOrigin {
 
 SDPOrigin sdp_get_origin(const std::string& sdp);
 
+// Copied from upstream cpp-httplib
+namespace httplib::detail {
+std::string encode_url(const std::string &);
+} // namespace httplib::detail
+
 #endif
